src/zkbob/ZkBobPool.sol

```diff
@@ -238,16 +242,24 @@ abstract contract ZkBobPool is IZkBobPool, EIP1967Admin, Ownable, Parameters, Zk
         if (txType == 0) {
             // Deposit
             require(transfer_token_delta > 0 && energy_amount == 0, "ZkBobPool: incorrect deposit amounts");
-            IERC20(token).safeTransferFrom(user, address(this), uint256(token_amount) * TOKEN_DENOMINATOR);
+            // Assuming token_amount is in the legacy nomination,
+            // amount of tokens to transfer must be calculated as
+            // token_amount * D, where
+            //   D = 10 ** (token.decimals() - 9)
+            IERC20(token).safeTransferFrom(user, address(this), poolToToken(uint256(token_amount), TOKEN_DENOMINATOR));
```

Comment:
If I deposit `1.000_000_999` zkBob units, contract will transfer `1.000_000` USDC units, resulting in a loss of `1e-6 USDC`. I think we should prohibit such deposits on the contract level.

The same is applicable for withdrawals, maybe it also make sense to limit them as well.

Reply:
addressed in 384ac22, take a look and if you have more elegant suggestion please let me know